src/app/settings/keyboard-mapping-settings/components/edit-keyboard-mapping/edit-keyboard-mapping.component.html

```diff
@@ -1,39 +1,65 @@
-                <ng-container [ngSwitch]="getSelectedActionArgumentType()">
-                    <sofie-text-input *ngSwitchCase="ActionArgumentSchemaType.STRING" [label]="selectedAction?.argument?.name!" [formControlName]="'actionArguments'" [helpText]="selectedAction?.argument?.description!"></sofie-text-input>
-                    <sofie-number-input *ngSwitchCase="ActionArgumentSchemaType.NUMBER" [label]="selectedAction?.argument?.name!" [formControlName]="'actionArguments'" [helpText]="selectedAction?.argument?.description!"></sofie-number-input>
-                </ng-container>
-            </ng-container>
-        </div>
-    </form>
-    <sofie-action-selector [selectedAction]="selectedAction" (actionSelected)="selectAction($event)" class="ps-5"></sofie-action-selector>
+      <ng-container *ngIf="doesSelectedActionHaveActionArguments()">
+        <ng-container [ngSwitch]="getSelectedActionArgumentType()">
+          <sofie-text-input
+            *ngSwitchCase="ActionArgumentSchemaType.STRING"
+            [label]="selectedAction?.argument?.name!"
+            [formControlName]="'actionArguments'"
+            [helpText]="selectedAction?.argument?.description!"
+            [isRequired]="true"
+            [errorMessage]="actionArgumentErrorMessage">
+          </sofie-text-input>
+          <sofie-number-input
+            *ngSwitchCase="ActionArgumentSchemaType.NUMBER"
+            [label]="selectedAction?.argument?.name!"
+            [formControlName]="'actionArguments'"
+            [helpText]="selectedAction?.argument?.description!"
+            [isRequired]="true"
+            [errorMessage]="actionArgumentErrorMessage">
+          </sofie-number-input>
+        </ng-container>
+      </ng-container>
+    </div>
+  </form>
+  <sofie-action-selector [selectedAction]="selectedAction" (actionSelected)="selectAction($event)" class="ps-5"> </sofie-action-selector>
 </sofie-row>
 <sofie-button-group>
-    <sofie-button (click)="cancel()" i18n>global.close.button</sofie-button>
-    <sofie-spacer></sofie-spacer>
-    <sofie-button (click)="save()" [type]="'PRIMARY'" [isDisabled]="!actionTriggerForm.valid" i18n>common.save</sofie-button>
+  <sofie-button (click)="cancel()" i18n>global.close.button</sofie-button>
+  <sofie-spacer></sofie-spacer>
+  <sofie-button (click)="save()" [type]="'PRIMARY'" [isDisabled]="!actionTriggerForm.valid" [tooltip]="saveButtonErrorsTooltip" i18n>common.save</sofie-button>
 </sofie-button-group>
 
 <ng-template #actionArguments>
-    <div class="selected-action mb-1">Selected action</div>
-    <sofie-row>
-        <sofie-action-box [action]="selectedAction!"></sofie-action-box>
-        <div class="selected-action-description ps-1">
-            {{ selectedAction!.description }}
-        </div>
-    </sofie-row>
+  <div class="selected-action mb-1"><ng-container i18n>action-triggers.selected-action.label</ng-container> <sofie-form-required-mark></sofie-form-required-mark></div>
```

Comment:
Please newline nested componenets. It's way to easy to miss components when they are formatted on one line.

Reply:
Fix